tools/cache_perf_tool/main.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2025-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package main // main.go
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"runtime/pprof"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/db"
+	"github.com/couchbaselabs/rosmar"
+)
+
+var numGoroutines atomic.Int32
+
+func main() {
+	mode := flag.String("mode", "processEntry", "Mode for the tool to run in, either dcp or processEntry.")
+	nodes := flag.Int("sgwNodes", 1, "Number of sgw nodes to abstract.")
+	bachSize := flag.Int("batchSize", 10, "Batch size for the sequence allocator.")
+	timeToRun := flag.Int("duration", 5, "Duration to run the test for in minutes.")
```

Comment:
```suggestion
	timeToRun := flag.Int("duration", 5, "Duration to run the test for in minutes.")
```

Switch to `time.Duration` to allow for lower values like `1s` for devlopment